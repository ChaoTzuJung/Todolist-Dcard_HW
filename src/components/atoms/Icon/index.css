.icon {
	color: #000;

	&.light {
		color: #fff;
	}
}
