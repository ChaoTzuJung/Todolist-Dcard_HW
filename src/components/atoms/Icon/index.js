import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const Icon = ({
	children,
	className,
	light = false,
	size = 'md-24', // CSS rules for the standard material design sizing guidelines (.md-18 .md-24 .md-36 .md-48)
	...other
}) => (
	<i
		className={classnames(`material-icons ${size}`, styles.icon, className, {
			[styles.light]: light,
		})}
		{...other}
	>
		{children}
	</i>
);

export default Icon;
