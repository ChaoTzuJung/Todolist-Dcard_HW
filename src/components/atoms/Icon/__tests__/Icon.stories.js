import React from 'react';
import { withKnobs, boolean } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';

import Icon from 'components/atoms/Icon';

const stories = storiesOf('Commons|atoms/Icon', module);

stories.addDecorator(withKnobs);

stories.add('__interactive', () => <Icon light={boolean('light', false)}>add</Icon>);

stories.add('outlined icon', () => (
	<Icon outlined={boolean('outlined', true)}>insert_drive_file</Icon>
));

const usedIcon = [
	'add',
	'date_range',
	'insert_drive_file',
	'textsms',
	'edit',
	'star',
	'star_border',
	'check_box',
	'check_box_outline_blank',
];

stories.add('used icon', () => (
	<div style={{ display: 'flex', flexWrap: 'wrap' }}>
		{usedIcon.map(icon => (
			<div
				key={icon}
				style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', margin: '20px' }}
			>
				<Icon>{icon}</Icon>
				<span>{icon}</span>
			</div>
		))}
	</div>
));
