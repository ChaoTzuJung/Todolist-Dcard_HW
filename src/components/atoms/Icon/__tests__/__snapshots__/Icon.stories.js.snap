// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Icon
  light={false}
>
  <i
    className="icon material-icons"
  >
    add
  </i>
</Icon>
`;

exports[`outlined icon 1`] = `
<Icon
  outlined={true}
>
  <i
    className="icon material-icons-outlined"
  >
    insert_drive_file
  </i>
</Icon>
`;

exports[`used icon 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
>
  <div
    key="add"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        add
      </i>
    </Icon>
    <span>
      add
    </span>
  </div>
  <div
    key="date_range"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        date_range
      </i>
    </Icon>
    <span>
      date_range
    </span>
  </div>
  <div
    key="insert_drive_file"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        insert_drive_file
      </i>
    </Icon>
    <span>
      insert_drive_file
    </span>
  </div>
  <div
    key="textsms"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        textsms
      </i>
    </Icon>
    <span>
      textsms
    </span>
  </div>
  <div
    key="edit"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        edit
      </i>
    </Icon>
    <span>
      edit
    </span>
  </div>
  <div
    key="star"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        star
      </i>
    </Icon>
    <span>
      star
    </span>
  </div>
  <div
    key="star_border"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        star_border
      </i>
    </Icon>
    <span>
      star_border
    </span>
  </div>
  <div
    key="check_box"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        check_box
      </i>
    </Icon>
    <span>
      check_box
    </span>
  </div>
  <div
    key="check_box_outline_blank"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "margin": "20px",
      }
    }
  >
    <Icon>
      <i
        className="icon material-icons"
      >
        check_box_outline_blank
      </i>
    </Icon>
    <span>
      check_box_outline_blank
    </span>
  </div>
</div>
`;
